In [2]:
import os
import prefect
import pandas as pd

In [1]:
!prefect version

Version:             3.4.4
API version:         0.8.4
Python version:      3.12.2
Git commit:          0367d7aa
Built:               Thu, May 29, 2025 09:37 PM
OS/Arch:             linux/x86_64
Profile:             ephemeral
Server type:         ephemeral
Pydantic version:    2.11.5
Server:
  Database:          sqlite
  SQLite version:    3.46.0


### Question 1. Select the Tool

What's the name of the orchestrator you chose? **prefect**

### Question 2. Version

What's the version of the orchestrator? **3.4.4**

### Question 3. Creating a pipeline

In [4]:
data_link = "https://d37ci6vzurychx.cloudfront.net/trip-data/yellow_tripdata_2023-03.parquet"
df = pd.read_parquet(data_link)
num_records = df.shape[0]
print(f"The number of records is {num_records}.")

The number of records is 3403766.


How many records did we load? **3,403,766**

### Question 4. Data preparation

In [9]:
os.system("python prefect_pipeline.py")

(3316216, 20)


0

What's the size of the result? **3,316,216**

### Question 5. Train a model

In [12]:
df.head()

VendorID tpep_pickup_datetime tpep_dropoff_datetime  passenger_count  \
0         2  2023-03-01 00:06:43   2023-03-01 00:16:43              1.0   
1         2  2023-03-01 00:08:25   2023-03-01 00:39:30              2.0   
2         1  2023-03-01 00:15:04   2023-03-01 00:29:26              0.0   
3         1  2023-03-01 00:49:37   2023-03-01 01:01:05              1.0   
4         2  2023-03-01 00:08:04   2023-03-01 00:11:06              1.0   

   trip_distance  RatecodeID store_and_fwd_flag  PULocationID  DOLocationID  \
0           0.00         1.0                  N           238            42   
1          12.40         1.0                  N           138           231   
2           3.30         1.0                  N           140           186   
3           2.90         1.0                  N           140            43   
4           1.23         1.0                  N            79           137   

   payment_type  fare_amount  extra  mta_tax  tip_amount  tolls_amount  \
0             2          8.6    1.0      0.5        0.00           0.0   
1             1         52.7    6.0      0.5       12.54           0.0   
2             1         18.4    3.5      0.5        4.65           0.0   
3             1         15.6    3.5      0.5        4.10           0.0   
4             1          7.2    1.0      0.5        2.44           0.0   

   improvement_surcharge  total_amount  congestion_surcharge  Airport_fee  
0                    1.0         11.10                   0.0         0.00  
1                    1.0         76.49                   2.5         1.25  
2                    1.0         28.05                   2.5         0.00  
3                    1.0         24.70                   2.5         0.00  
4                    1.0         14.64                   2.5         0.00

In [16]:
os.system("python prefect_pipeline.py")

24.77448997808052


0

What's the intercept of the model? **24.77**

### Question 6. Register the model

In [3]:
os.system("python prefect_pipeline.py")

2025/06/01 17:25:20 INFO mlflow.tracking.fluent: Experiment with name 'orchestration_experiment' does not exist. Creating a new experiment.
2025/06/01 17:25:59 WARNING mlflow.models.model: Model logged without a signature and input example. Please set `input_example` parameter when logging the model to auto infer the model signature.
Successfully registered model 'linear-reg-model'.
2025/06/01 17:26:00 INFO mlflow.store.model_registry.abstract_store: Waiting up to 300 seconds for model version to finish creation. Model name: linear-reg-model, version 1
Created version '1' of model 'linear-reg-model'.


🏃 View run useful-shrew-495 at: http://127.0.0.1:5000/#/experiments/1/runs/060605e97a534c78836ff4cc8494ccaa
🧪 View experiment at: http://127.0.0.1:5000/#/experiments/1
Finished training and registering run_id=060605e97a534c78836ff4cc8494ccaa as 'linear-reg-model'.


0

model_size_bytes: **4500**

In [4]:
os.system("python prefect_pipeline.py")

17:36:10.088 | INFO    | prefect - Starting temporary server on http://127.0.0.1:8722
See https://docs.prefect.io/3.0/manage/self-host#self-host-a-prefect-server for more information on running a dedicated Prefect server.
17:36:13.487 | INFO    | Flow run 'terrestrial-leopard' - Beginning flow run 'terrestrial-leopard' for flow 'main-flow'
17:36:17.531 | INFO    | Task run 'read_dataframe-39c' - Finished in state Completed()
2025/06/01 17:36:53 WARNING mlflow.models.model: Model logged without a signature and input example. Please set `input_example` parameter when logging the model to auto infer the model signature.
17:36:53.492 | INFO    | Task run 'train_model-cfe' - Finished in state Completed()
Registered model 'linear-reg-model' already exists. Creating a new version of this model...
2025/06/01 17:36:53 INFO mlflow.store.model_registry.abstract_store: Waiting up to 300 seconds for model version to finish creation. Model name: linear-reg-model, version 2
Created version '2' of mod

🏃 View run resilient-stag-878 at: http://127.0.0.1:5000/#/experiments/1/runs/d5f616d5acef460fa481cd56d8ae105f
🧪 View experiment at: http://127.0.0.1:5000/#/experiments/1
Finished training and registering run_id=d5f616d5acef460fa481cd56d8ae105f as 'linear-reg-model'.


0